bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/StateFilterProfile.java

```diff
@@ -173,47 +180,177 @@ State parseState(@Nullable String stateString) {
+                                if (!(parsedValue instanceof QuantityType)) {
+                                    parsedValue = parsedValue.as(state.getClass());
+                                }
+                                logger.debug("Condition value: '{}' is an item state: '{}' ({})", value, parsedValue,
+                                        parsedValue == null ? "null" : parsedValue.getClass().getSimpleName());
+                            }
+                        } catch (ItemNotFoundException ignore) {
+                        }
+                    }
+
+                    if (parsedValue == null) {
+                        if (comparisonType == ComparisonType.NEQ) {
+                            // They're not even type compatible, so return true for NEQ comparison
+                            return true;
+                        } else {
+                            logger.debug("Condition value: '{}' is not compatible with state '{}' ({})", value, state,
+                                    state.getClass().getSimpleName());
+                            return false;
+                        }
+                    }
+                }
+
+                rhs = Objects.requireNonNull(parsedValue instanceof StringType ? parsedValue.toString() : parsedValue);
+
+                if (logger.isDebugEnabled()) {
+                    if (itemName.isEmpty()) {
+                        logger.debug("Performing a comparison between input '{}' ({}) and value '{}' ({})", lhs,
+                                lhs.getClass().getSimpleName(), rhs, rhs.getClass().getSimpleName());
+                    } else {
+                        logger.debug("Performing a comparison between item '{}' state '{}' ({}) and value '{}' ({})",
+                                itemName, lhs, lhs.getClass().getSimpleName(), rhs, rhs.getClass().getSimpleName());
+                    }
                 }
-                default:
-                    logger.warn("Unknown condition type {}. Expected 'eq' or 'neq' - skipping state update",
-                            comparisonType);
-                    return false;
 
+                return switch (comparisonType) {
+                    case EQ -> lhs.equals(rhs);
+                    case NEQ -> !lhs.equals(rhs);
+                    case GT -> ((Comparable) lhs).compareTo(rhs) > 0;
+                    case GTE -> ((Comparable) lhs).compareTo(rhs) >= 0;
+                    case LT -> ((Comparable) lhs).compareTo(rhs) < 0;
+                    case LTE -> ((Comparable) lhs).compareTo(rhs) <= 0;
+                };
+            } catch (ItemNotFoundException | IllegalArgumentException | ClassCastException e) {
+                logger.warn("Error evaluating condition: {}: {}", this, e.getMessage());
             }
+            return false;
         }
 
         enum ComparisonType {
-            EQ,
-            NEQ
+            EQ("=="),
+            NEQ("!="),
```

Comment:
What about `<>`? IMO this is not uncommon for "not equal". Maybe you can add `NEQ_ALT("<>")` and also add `NEQ_ALT` to the `NEQ` case above.

Reply:
I've added `<>`